ow_plexil/scripts/identify_sample_location.py

```diff
@@ -0,0 +1,243 @@
+          self.largest_area = sample_point_area
+          self.best_sample_location = sample_point_3d
+          self.sample_location_image = cv_image
+          self.uv = uv
+
+      #if succesful we return our point, otherwise we return an empty list
+      if self.best_sample_location == None:
+        rospy.loginfo("Could not find valid sample location")
+        self.sample_location_action_server.set_succeeded(IdentifyLocationResult(success=False, sample_location=None))
+      else:
+        #publishes the modified image showing our sample location choice
+        self.publish_chosen_location_image()
+        self.visualize_sample_point(self.best_sample_location)
+        self.sample_location_action_server.set_succeeded(IdentifyLocationResult(success=True, sample_location=self.best_sample_location ))
+
+    else:
+      rospy.loginfo("No images have been recorded")
+      self.sample_location_action_server.set_succeeded(IdentifyLocationResult(success=False, sample_location=None))
+
+
+  def site_imaging_callback(self, rect_img_msg, points2_msg):
+    '''Synchronizes the pointcloud and the rectified image topic and saves them as a tuple to the image_history list.
+    Starts to delete old messages at a length of 50.'''
+    #get both messages and save them to our image history for later processing
+    self.image_history.append((rect_img_msg, points2_msg))
+
+    #prevent our image history from getting too large saves most recent 50 images
+    if len(self.image_history) > 50:
+      self.image_history = self.image_history[-50:]
+
+
+  def publish_chosen_location_image(self):
+    '''Publishes the original image with our drawn on contours and chosen sample location. 
+    Used to visualize the chosen sample point.'''
+    #draws a circle where our chosen sample location is located
+    cv.circle(self.sample_location_image, self.uv, 7, (0,255,0),-1)
+    #converts back to a ros image msg before publishing
+    try:
+      image = self.bridge.cv2_to_imgmsg(self.sample_location_image, "bgr8")
+    except CvBridgeError as err:
+      rospy.logerror("CV Bridge error: {0}".format(err))
+      return
+    self.publish_photo.publish(image)
+
+
+  def identify_sample_spots(self, location_image, filter_type):
+    '''Uses OpenCv to process the given image. For filter type Brown we isolate a range of brown colors 
+    on the image and draw the contours. For Dark filter (the default filter) we isolate the darkest spots 
+    in the image (usually shadows) and find the countours. The 2d center location is taken for these contours
+    and are sorted in descending order, we then draw/store the largest 5 contours/center points to return.'''
+    try:
+      image = self.bridge.imgmsg_to_cv2(location_image, "bgr8")
+    except CvBridgeError as err:
+      rospy.logerror("CV Bridge error: {0}".format(err))
+      return None, None, None
+
+    #defaults to Dark Spot if argument doesnt match or none is given
+    if filter_type == "Brown":
```

Comment:
Optional, consider this:
```suggestion
    if filter_type.lower() == "brown":
```